src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,100 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_CAPACITY = 16;
+    private static final int DEFAULT_GROW = 2;
+    private static final float LOAD_FACTOR = 0.75f;
+
+    private Node<K, V>[] table;
+    private int size;
+    private int capacity;
+
+    public MyHashMap() {
+        table = ( Node<K, V>[]) new Node[DEFAULT_CAPACITY];
+        capacity = table.length;
+    }
 
     @Override
     public void put(K key, V value) {
-
+        int bucket = key == null ? 0 : key.hashCode() % capacity;
+        bucket = Math.abs(bucket);
+        Node<K, V> currentNode = table[bucket];
+        while (currentNode != null) {
+            if (Objects.equals(key, currentNode.key)) {
+                currentNode.value = value;
+                return;
+            }
+            currentNode = currentNode.next;
+        }
+        Node<K, V> newNode = new Node<>(key, value,table[bucket]);
+        table[bucket] = newNode;
+        size++;
+        if (size >= capacity * LOAD_FACTOR) {
+            resize();
+        }
     }
 
     @Override
     public V getValue(K key) {
+        int bucket = key == null ? 0 : key.hashCode() % capacity;
+        bucket = Math.abs(bucket);
+        Node<K, V> currentNode = table[bucket];
```

Comment:
This looks like you use a similar code in the put method. Mb you should create a separate method.